Client/Frontend/Browser/BrowserViewController/BrowserViewController+Menu.swift

```diff
@@ -211,6 +169,74 @@ extension BrowserViewController {
       }
     }
   }
+  
+  private func bookmarksButton(_ menuController: MenuViewController) -> MenuItemButton {
+    MenuItemButton(icon: UIImage(named: "menu_bookmarks", in: .current, compatibleWith: nil)!.template, title: Strings.bookmarksMenuItem) { [unowned self, unowned menuController] in
+      let vc = BookmarksViewController(
+        folder: bookmarkManager.lastVisitedFolder(),
+        bookmarkManager: bookmarkManager,
+        isPrivateBrowsing: PrivateBrowsingManager.shared.isPrivateBrowsing)
+      vc.toolbarUrlActionsDelegate = self
+      menuController.presentInnerMenu(vc)
+    }
+  }
+  
+  private func historyButton(_ menuController: MenuViewController) -> MenuItemButton {
```

Comment:
Swap all these MenuButtonItem creations with a single function similar to 

```
  private func createMenuButton(for buttonType: MenuButtonType, completion: @escaping () -> Void) -> MenuItemButton {   
    MenuItemButton(icon: buttonType.icon, title: buttonType.title, subTitle: buttonType.subTitle) {
      completion()
    }
  }
```

Reply:
Hi @soner-yuksel, thanks for the review! I like this idea of moving the button to some unified place. Instead of using a function for that, what do you think about moving this logic to a separate component? If we do that, we can still have all the benefits that you mention, but it will also be much cleaner the code for both this new component and the current file. More important, it will also be much easy to add unit tests. I thought something like that: 

```
class MenuButtonFactory {
    enum MenuButtonType {
      case news
      case talk
      case wallet(subtitle: String)
      
      var icon: UIImage {
        switch self {
          case .news:
            return UIImage(named: "menu_brave_news", in: .current, compatibleWith: nil)!.template
          case .talk:
            return UIImage(named: "menu-brave-talk", in: .current, compatibleWith: nil)!.template
          case .wallet(_):
            return UIImage(named: "menu-crypto", in: .current, compatibleWith: nil)!.template
        }
      }
      
      var title: String {
        switch self {
        case .news:
          return ""
        case .talk:
          return ""
        case .wallet:
          return ""
        }
      }
      
      var subtitle: String {
        switch self {
        case .news:
          return ""
        case .talk:
          return ""
        case .wallet:
          return ""
        }
      }
    }
    
    static func createMenuButton(for buttonType: MenuButtonType, completion: @escaping () -> Void) -> MenuItemButton {
      MenuItemButton(icon: buttonType.icon, title: buttonType.title, subtitle: buttonType.subtitle) {
        completion()
      }
    }
  }
```

```
 MenuButtonFactory.createMenuButton(for: .news)
```

What do you think?

Reply:
@mauriciomaniglia  Sounds like a good idea. Please add the factory class and keep the completion handling wher ethe button is created.

Reply:
@soner-yuksel I've just updated the PR with the changes

Reply:
@mauriciomaniglia thanks you for all the changes, I have one more comment but after that it looks like it ready to go.